src/components/pages/DataVisualizations/Graphs/CitizenshipMapAll.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import Plot from 'react-plotly.js';
+import { connect } from 'react-redux';
+import Table from './TableComponents/Table';
+import { colors } from '../../../../styles/data_vis_colors';
+
+const { background_color, secondary_accent_color } = colors;
+
+const mapStateToProps = state => {
+  return {
+    citizenshipMapAllData: state.vizReducer.citizenshipMapAllData,
+  };
+};
+
+function CitizenshipMapAll(props) {
+  const { citizenshipMapAllData } = props;
+  const [plotlyGraphAxis, setPlotlyGraphAxis] = useState({
+    locationsAndText: [],
+    z: [],
+  });
+
+  useEffect(() => {
+    if (citizenshipMapAllData['countryGrantRateObj'] !== undefined) {
+      setPlotlyGraphAxis({
+        locationsAndText:
+          citizenshipMapAllData['countryGrantRateObj']['countries'],
+        z: citizenshipMapAllData['countryGrantRateObj'][
+          'countriesPercentGranteds'
+        ],
+      });
+    } else {
+      setPlotlyGraphAxis({ locationsAndText: [], z: [] });
+    }
+  }, [citizenshipMapAllData]);
+
+  const geoScopeArray = [
+    'world',
+    'usa',
+    'europe',
+    'asia',
+    'africa',
+    'north america',
+    'south america',
+  ];
+  const [geoScope, setGeoScope] = useState('world');
+  const handleScopeChange = e => {
+    //update Plotly region based on dropdown selection
+    const { value } = e.target;
+    setGeoScope(value);
+  };
+
+  const rowsForTable = citizenshipMapAllData.hasOwnProperty('rowsForTable')
```

Comment:
When you use `.hasOwnProperty` how do you expect to unit this? With proptypes, unit testing is pretty straight forward. Why is `.hasOwnProperty` better than that?

Reply:
Hi Ash, I rewrote the .hasOwnProperty lines to useState and useEffect instead. Originally the .hasOwnProperty was used in a ternary expression to check if it exists within the props object coming from Redux Store however, it is now using useState/effect. Each graph component will have the row data set to an empty array on render if there's no associated rowData key to prevent page from crashing due to undefined data and will be updated once user submits a query.

Reply:
Makes sense! Thank you for clarifying your thought process.